simulator/mips/t/unit_test.cpp

```diff
@@ -236,10 +146,2252 @@ TEST_CASE( "MIPS32_instr_disasm: Process_Disasm_Branches")
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0xa1);
+}
+
+TEST_CASE( "MIPS32_instr: srlv test_2")
+{
+    MIPS32Instr instr( "srlv");
+    instr.set_v_src( 153, 0);
+    instr.set_v_src( 3, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 19);
+}
+
+TEST_CASE( "MIPS32_instr: srlv test_3")
+{
+    MIPS32Instr instr( "srlv");
+    instr.set_v_src( 0xfeed, 0);
+    instr.set_v_src( 8, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0x00fe);
+}
+
+TEST_CASE( "MIPS32_instr: srlv test_4")
+{
+    MIPS32Instr instr( "srlv");
+    instr.set_v_src( 0xaaa00000, 0);
+    instr.set_v_src( 4, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0x0aaa0000);
+}
+
+// Bug-test (#709)
+/* ***********************************
+TEST_CASE( "MIPS32_instr: srlv test_5")
+{
+    MIPS32Instr instr( "srlv");
+    instr.set_v_src( 1, 0);
+    instr.set_v_src( 64, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 1);
+}
+// ***********************************/
+
+// Bug-test (#709)
+/* ***********************************
+TEST_CASE( "MIPS64_instr: srlv test_1")
+{
+    MIPS64Instr instr( "srlv");
+    instr.set_v_src( 0x11, 0);
+    instr.set_v_src( 0x00000a00, 1);
+    instr.execute();
+    CHECK( instr.get_v_dst() == 0x11);
+}
+// ***********************************/
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: syscall test_1")
+{
+    CHECK(MIPS32Instr(0x0000000c).get_disasm() == "syscall");
+    
+    MIPS32Instr instr( "syscall");
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::SYSCALL);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: teq test_eq")
+{
+    CHECK(MIPS32Instr(0x02290034).get_disasm() == "teq $s1, $t1");
+    
+    MIPS32Instr instr( "teq");
+    instr.set_v_src( 15, 0);
+    instr.set_v_src( 15, 1);
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::EXPLICIT_TRAP);
+}
+
+TEST_CASE( "MIPS32_instr: teq test_ne")
+{
+    MIPS32Instr instr( "teq");
+    instr.set_v_src( 0xff, 0);
+    instr.set_v_src( 0, 1);
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::NO_TRAP);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: teqi test_eq")
+{
+    CHECK(MIPS32Instr(0x062c04d2).get_disasm() == "teqi $s1, 0x4d2");
+    CHECK(MIPS32Instr(0x062cfb2e).get_disasm() == "teqi $s1, 0xfb2e");
+    
+    MIPS32Instr instr( "teqi");
+    instr.set_v_src( 0, 0);
+    instr.set_v_imm( 0);
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::EXPLICIT_TRAP);
+}
+
+TEST_CASE( "MIPS32_instr: teqi test_ne")
+{
+    MIPS32Instr instr( "teqi");
+    instr.set_v_src( 0, 0);
+    instr.set_v_imm( 4);
+    instr.execute();
+    CHECK( instr.trap_type() == Trap::NO_TRAP);
+}
+////////////////////////////////////////////////////////////////////////////////
+
+TEST_CASE( "MIPS32_instr: tge test_l")
```

Comment:
Please expand `test_l` to something more verbose, like `test if less`. You can easily do it with substitution.

Reply:
Done.